src/app/signup/page.tsx

```diff
@@ -1,7 +1,53 @@
+'use client';
+
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { signup } from '@/src/app/api';
+import { useRouter } from 'next/navigation';
+
 const Signup = () => {
+  const [name, setName] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [successMessage, setSuccessMessage] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const router = useRouter();
+
+  const handleState = (e: ChangeEvent<HTMLInputElement>, setState: React.Dispatch<React.SetStateAction<string>>) => {
+    setState(e.target.value);
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setIsLoading(true);
+
+    try {
+      const data = await signup({ email, password, name });
+
+      router.push('/login');
+    } catch (error: unknown) {
+      console.error(error);
+      setErrorMessage(`${error}`);
```

Comment:
Take a look in this package. It's super great and easy to use[React Toastify](https://www.npmjs.com/package/react-toastify)